src/main/java/org/traccar/protocol/ArknavProtocolDecoder.java

```diff
@@ -59,23 +62,40 @@ protected Object decode(
             return null;
         }
 
-        Position position = new Position(getProtocolName());
-
         DeviceSession deviceSession = getDeviceSession(channel, remoteAddress, parser.next());
         if (deviceSession == null) {
             return null;
         }
+        
+        Position position = new Position(getProtocolName());
         position.setDeviceId(deviceSession.getDeviceId());
-
+        long status = parser.nextHexLong();
+        String model = parser.next();
         position.setValid(parser.next().equals("A"));
         position.setLatitude(parser.nextCoordinate());
         position.setLongitude(parser.nextCoordinate());
-        position.setSpeed(parser.nextDouble(0));
+        position.setSpeed(UnitsConverter.kphFromKnots(parser.nextDouble(0)));
         position.setCourse(parser.nextDouble(0));
-
         position.set(Position.KEY_HDOP, parser.nextDouble(0));
-
         position.setTime(parser.nextDateTime(Parser.DateTimeFormat.HMS_DMY));
+        if (model.contains("PT33")) {
+            position.set(Position.KEY_BATTERY_LEVEL, parser.nextInt());
+            if ((status & 0x100L) != 0) {
+                position.addAlarm(Position.ALARM_LOW_BATTERY);
+            }
+            if ((status & 0x200L) != 0) {
+                position.addAlarm(Position.ALARM_MOVEMENT);
+            }
+            if ((status & 0x400L) != 0) {
+                position.addAlarm(Position.ALARM_GEOFENCE_EXIT);
+            }
+            if ((status & 0x010L) != 0) {
+                position.addAlarm(Position.ALARM_SOS);
+            }
+            if ((status & 0x001L) != 0) {
+                position.addAlarm(Position.ALARM_OVERSPEED);
+            }
```

Comment:
Where can I find this in the doc?

Reply:
https://github.com/user-attachments/files/17003217/personal.tracker.manual.V2.pdf
pag 39

Reply:
It looks like the value is not HEX, but a specially encoded 3 numbers.